### ISSUE TITLE
Training stuck

### ISSUE THREAD

**Issue by Madhavan0123** [p0p4k/vits2_pytorch#71]
Hello ,

Thanks for all the effort to create this repo. When I launch training it runs for a few steps and then I see no progress at all. Its just stuck without any progress for a long time. It still hasnt progressed. 
INFO:ljs_base:Saving model and optimizer state at iteration 1 to ./logs/ljs_base/G_0.pth
INFO:ljs_base:Saving model and optimizer state at iteration 1 to ./logs/ljs_base/D_0.pth
INFO:ljs_base:Saving model and optimizer state at iteration 1 to ./logs/ljs_base/DUR_0.pth
Loading train data:   4%|████████████▍   

have you encountered this before ? Any help would be extremely helpful

**Comment by p0p4k** [comment]
Hi, temporarily turn off duration discriminator and tell me if it works.

**Comment by Madhavan0123** [comment]
Yes it seems to be working for now.  Any reason with the duration discriminator is causing the issue ?

**Comment by p0p4k** [comment]
I feel my implementation was too naive. Might need to correct it with some testing. Busy on other models now, will do it when I have some time on me. Let me know about the audio quality after you train. Thanks.

**Comment by LilHOliang** [comment]
Hello,thank you for your great effort !
I meet the same problem and I want to know will you correct it recently or still busy on other models?

**Comment by p0p4k** [comment]
I have moved to improving `pflowtts`.

**Comment by JohnHerry** [comment]
> I have moved to improving `pflowtts`.

Hi, p0p4k,
     How is the pflowtts growing now? is it a better choice then vits2？ can it support both normal tts and zero-shot tts?

**Comment by p0p4k** [comment]
I think better than vits/vits2. Only downside it not being e2e.

**Comment by JohnHerry** [comment]
> I think better than vits/vits2. Only downside it not being e2e.

ok, thank you.

**Comment by codeghees** [comment]
@p0p4k do you know the bug here?

**Comment by p0p4k** [comment]
@codeghees which part ? training stuck part?

**Comment by codeghees** [comment]
yep

**Comment by codeghees** [comment]
In the same boat.

**Comment by p0p4k** [comment]
@codeghees did not look into this personally cause of no gpu yet. Maybe you can try to debug and send a PR. I can assist you. Thanks a lot!

**Comment by codeghees** [comment]
Yep, will do! Trying to debug this.

**Comment by codeghees** [comment]
@p0p4k bug is on line
`scaler.scale(loss_gen_all).backward()`

Seems like GradScalar has issues with multi-gpu. I removed it and replaced it with standard backprop. The issue persists. Looks like a multi GPU issue.

**Comment by p0p4k** [comment]
Works on single gpu?

**Comment by codeghees** [comment]
Haven't tested yet. Trying a run with fp16 enabled.

**Comment by farzanehnakhaee70** [comment]
@p0p4k I have no issues for single GPU training. But it will stuck if I do multiple GPU training. Any success for resolving the issue?
